backend/routes/quests.go

```diff
@@ -260,3 +244,137 @@ func ClaimTodayQuestDevnet(w http.ResponseWriter, r *http.Request) {
 
 	routeutils.WriteResultJson(w, "Today quest claimed")
 }
+
+func SendChat(w http.ResponseWriter, r *http.Request) {
+	userAddress := "0000000000000X432"
+
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	chat, err := routeutils.ReadJsonBody[FactionChatRequest](r)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Failed to parse request body")
+		return
+	}
+
+	type ExistsResult struct {
+		Exists int `db:"exists"`
+	}
+
+	result, err := core.PostgresQueryOne[ExistsResult](`SELECT 1 FROM FactionMembersInfo WHERE faction_i = $1 AND user_address = $2 LIMIT 1`, chat.FactionKey, userAddress)
+
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Failed to parse request body")
+		return
+	}
+
+	if result != nil {
+		// Insert chat into database
+		err = core.PostgresExec(r.Context(), "INSERT INTO FactionChats (sender, factionKey, message) VALUES ($1, $2, $3)", userAddress, chat.FactionKey, chat.Message)
+		if err != nil {
+			routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to send chat")
+			return
+		}
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	routeutils.WriteResultJson(w, "FactionChat updated successfully")
+}
+
+func DeleteChat(w http.ResponseWriter, r *http.Request) {
+	userAddress := "0000000000000X432"
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	// Parse query params
+	chat, err := routeutils.ReadJsonBody[FactionChatRequest](r)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Failed to parse request body")
+		return
+	}
+	if chat.FactionKey == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing chat ID parameter")
+		return
+	}
+
+	canDelete := false
+
+	// Retrieve the faction leader from the database
+	leader, err := core.PostgresQueryOne[string]("SELECT ftns.leader FROM Factions ftns INNER JOIN FactionChats ftcs ON ftns.faction_key = ftcs.key WHERE ftcs.id = $1", chat.FactionKey)
+	if err != nil {
+		http.Error(w, "Failed to retrieve faction leader", http.StatusInternalServerError)
+		return
+	}
+
+	if *leader == userAddress {
+		canDelete = true
+	}
+
+	if !canDelete {
```

Comment:
please merge any changes made into your other PR and fix all the changes mentioned there.